druid/src/widget/widget_ext.rs

```diff
@@ -125,8 +125,38 @@ pub trait WidgetExt<T: Data>: Widget<T> + Sized + 'static {
 
     /// Wrap this widget in a [`LensWrap`] widget for the provided [`Lens`].
     ///
+    /// Lenses are useful whenever a widget only needs access to a subfield of a larger struct or
+    /// generally access to part of a larger value.
+    /// For example: If one wants to embed a [`TextBox`] in a Widget of a Data type different from
+    /// `Widget<String>` they need to specify how to access a String within the Data.
+    ///
     /// [`LensWrap`]: ../struct.LensWrap.html
     /// [`Lens`]: ../trait.Lens.html
+    /// [`TextBox`]: ../TextBox.html
+    ///
+    ///
+    /// ```
+    /// use druid::{Data, Lens, Widget, widget::{TextBox, Flex, WidgetExt}};
+    ///
+    /// #[derive(Clone, Debug, Data, Lens)]
+    /// struct MyState {
+    ///     search_term: String,
+    ///     scale: f64,
+    ///     // ...
+    /// }
+    ///
+    ///
+    /// fn my_sidebar() -> impl Widget<MyState> {
+    ///     // `TextBox` is of type `Widget<String>`
+    ///     // via `.lens` we get it to be of type `Widget<MyState>`
+    ///     let searchbar = TextBox::new().lens(MyState::search_term);
```

Comment:
I would add a comment here that says something like,

"`MyState::search_term` is a lens generated by the `derive(Lens)` macro, that provides access to the `search_term` field."

Reply:
Sure, I overlooked this, but it's the most important part of the example. I extended the comment